frontend/src/components/SplashPage.js

```diff
@@ -1,32 +1,21 @@
 import React from 'react';
-
+import { useNavigate } from 'react-router-dom';
 import '../styles/SplashPage.css';
-import { getAuth, signInWithPopup, GoogleAuthProvider, GithubAuthProvider, createUserWithEmailAndPassword } from 'firebase/auth';
-import { initializeApp } from 'firebase/app';
 import app from '../firebaseConfig';
+import { getAuth, signInWithPopup, GoogleAuthProvider, GithubAuthProvider } from 'firebase/auth';
 
 const SplashPage = () => {
+  const navigate = useNavigate();
 
   const handleGoogleSignIn = () => {
     const auth = getAuth();
     const provider = new GoogleAuthProvider();
     signInWithPopup(auth, provider)
       .then((result) => {
-        // This gives you a Google Access Token. You can use it to access the Google API.
-        // const credential = GoogleAuthProvider.credentialFromResult(result);
-        // const token = credential.accessToken;
-        // The signed-in user info.
-        // const user = result.user;
-        // ...
+        // After sign-in, navigate to the UserHomepage
```

Comment:
HTTP POST request to backend goes here - first to auth, then register if they are new.

If register, then form for username should pop up asking them to choose a username

Reply:
Awesome, thank you Giovanni. Really excited to start incorporating these towards an initial prototype.